apps/prairielearn/src/pages/administratorWorkspaces/administratorWorkspaces.sql

```diff
@@ -1,37 +1,60 @@
--- BLOCK select_workspaces
+-- BLOCK select_workspace_hosts
+WITH
+  workspaces_with_context AS (
+    SELECT
+      w.id,
+      w.state,
+      w.workspace_host_id,
+      format_interval (
+        CASE
+          WHEN w.state = 'launching' THEN (NOW() - w.launched_at)
+          ELSE (NOW() - w.running_at)
+        END
+      ) AS time_in_state,
+      q.qid AS question_name,
+      ci.short_name AS course_instance_name,
+      c.short_name AS course_name,
+      i.short_name AS institution_name
+    FROM
+      workspaces AS w
+      JOIN variants AS v ON (w.id = v.workspace_id)
+      JOIN questions AS q ON (v.question_id = q.id)
+      LEFT JOIN course_instances AS ci ON (v.course_instance_id = ci.id)
+      JOIN pl_courses AS c ON (v.course_id = c.id)
+      JOIN institutions AS i ON (c.institution_id = i.id)
+    WHERE
+      w.state IN ('launching', 'running')
```

Comment:
We don't currently have an index on `state` for `workspaces` so I think this CTE will do a full table scan.

Reply:
Don't we?

https://github.com/PrairieLearn/PrairieLearn/blob/3cd0f83fa09d53d15c6b2c599b1c6f4f34be1aa4/database/tables/workspaces.pg#L25

Reply:
Jeez, and I checked this a bunch of times because I was surprised our old query wasn't horribly expensive. Ignore my noise!